packages/web-components/fast-foundation/src/tree-item/tree-item.ts

```diff
@@ -0,0 +1,269 @@
+                break;
+            case keyCodeArrowRight:
+                this.handleArrowRight();
+                break;
+            case keyCodeArrowDown:
+                // preventDefault to ensure we don't scroll the page
+                e.preventDefault();
+                this.focusNextNode(1);
+                break;
+            case keyCodeArrowUp:
+                // preventDefault to ensure we don't scroll the page
+                e.preventDefault();
+                this.focusNextNode(-1);
+                break;
+            case keyCodeEnter:
+                this.handleSelected(e);
+                break;
+            case keyCodeSpace:
+                this.handleSpaceBar();
+                break;
+        }
+
+        return true;
+    };
+
+    public handleExpandCollapseButtonClick = (): void => {
+        if (!this.disabled) {
+            this.setExpanded(!this.expanded);
+        }
+    };
+
+    public handleContainerClick = (e: MouseEvent): void => {
+        const expandButton: HTMLElement | null = this.expandCollapseButton;
+        if (
+            (!isHTMLElement(expandButton) ||
+                (isHTMLElement(expandButton) && expandButton !== e.target)) &&
+            !this.disabled
+        ) {
+            this.handleSelected(e);
+        }
+    };
+
+    private handleArrowLeft(): void {
+        if (this.expanded) {
+            this.setExpanded(false);
+        } else if (isHTMLElement(this.treeItem.parentElement)) {
+            const parentElement: HTMLElement | null = this.parentElement;
+
+            if (isHTMLElement(parentElement)) {
+                const parentNode: Element | null | undefined = parentElement!.closest(
+                    "[role='treeitem']"
+                );
+
+                if (isHTMLElement(parentNode)) {
+                    (parentNode as HTMLElement).focus();
+                }
+            }
+        }
+    }
+
+    private handleArrowRight(): void {
+        if (typeof this.expanded !== "boolean") {
+            return;
+        }
+
+        if (!this.expanded) {
+            this.setExpanded(true);
+        } else {
+            this.focusNextNode(1);
+        }
+    }
+
+    private handleSpaceBar(): void {
+        if (typeof this.expanded !== "boolean") {
+            return;
+        }
+        this.setExpanded(!this.expanded);
+    }
+
+    private focusNextNode(delta: number): void {
+        const visibleNodes: HTMLElement[] | void = this.getVisibleNodes();
+        if (!visibleNodes) {
+            return;
+        }
+
+        const currentIndex: number = visibleNodes.indexOf(this.treeItem);
+        const deltaPositive: boolean = delta >= 0;
+        if (currentIndex !== -1) {
+            let nextElement: HTMLElement = visibleNodes[currentIndex + delta];
+            while (nextElement.hasAttribute("disabled")) {
+                delta = deltaPositive ? delta + 1 : delta - 1;
+                nextElement = visibleNodes[currentIndex + delta];
```

Comment:
seems like you could skip modifying delta and having a boolean (deltaPositive) and just do:
```typescript
// instead of deltaPositive
const offset: number = delta >= 0 ? 1 : -1;
...
nextElement = visibleNodes[currentIndex + delta + offset]
```

Reply:
done and tested, much cleaner